creator-node/src/routes/files.js

```diff
@@ -484,6 +486,50 @@ module.exports = function (app) {
    */
   app.get('/ipfs/:dirCID/:filename', getDirCID)
 
+  /**
+   * Serves information on existence of given cids
+   * @param req
+   * @param req.body
+   * @param {string} req.body.cids the cids to check existence for
+   * @dev This route can have a large number of CIDs as input, therefore we use a POST request.
+   * TODO: Remove support for directories
+   */
+  app.post('/batch_cids_exist', handleResponse(async (req, res) => {
+    const { cids } = req.body
+
+    if (cids.length > BATCH_CID_ROUTE_LIMIT) {
```

Comment:
we should check that cids exists before calling .length. otherwise this throws an error